pkg/controller/containerjfr/containerjfr_controller_test.go

```diff
@@ -0,0 +1,462 @@
+
+	"github.com/operator-framework/operator-sdk/pkg/log/zap"
+	rhjmcv1beta1 "github.com/rh-jmc-team/container-jfr-operator/pkg/apis/rhjmc/v1beta1"
+	"github.com/rh-jmc-team/container-jfr-operator/pkg/controller/containerjfr"
+	"github.com/rh-jmc-team/container-jfr-operator/pkg/controller/containerjfr/resource_definitions"
+	"github.com/rh-jmc-team/container-jfr-operator/test"
+)
+
+var _ = Describe("ContainerjfrController", func() {
+	var (
+		objs       []runtime.Object
+		client     client.Client
+		controller *containerjfr.ReconcileContainerJFR
+	)
+
+	JustBeforeEach(func() {
+		logf.SetLogger(zap.Logger())
+		s := test.NewTestScheme()
+
+		client = fake.NewFakeClientWithScheme(s, objs...)
+		controller = &containerjfr.ReconcileContainerJFR{
+			Client:        client,
+			Scheme:        s,
+			ReconcilerTLS: test.NewTestReconcilerTLS(client),
+		}
+	})
+
+	AfterEach(func() {
+		objs = nil
+	})
+
+	Describe("reconciling a request", func() {
+		Context("succesfully creates required resources", func() {
+			BeforeEach(func() {
+				objs = []runtime.Object{
+					test.NewContainerJFR(),
+				}
+			})
+			It("should create certificates", func() {
+				expectCertificates(client, *controller)
+			})
+			It("should create routes", func() {
+				expectRoutes(client, *controller, false)
+			})
+			It("should create persistent volume claim and set owner", func() {
+				expectPVC(client, *controller, false)
+			})
+			It("should create Grafana secret and set owner", func() {
+				secret := &corev1.Secret{}
+				err := client.Get(context.Background(), types.NamespacedName{Name: "containerjfr-grafana-basic", Namespace: "default"}, secret)
+				Expect(err).To(HaveOccurred())
+
+				reconcileFully(client, *controller, false)
+
+				err = client.Get(context.Background(), types.NamespacedName{Name: "containerjfr-grafana-basic", Namespace: "default"}, secret)
+				Expect(err).ToNot(HaveOccurred())
+
+				// Compare to desired spec
+				expectedSecret := resource_definitions.NewGrafanaSecretForCR(test.NewContainerJFR())
+				checkMetadata(&secret.ObjectMeta, &expectedSecret.ObjectMeta)
+				Expect(secret.StringData["GF_SECURITY_ADMIN_USER"]).To(Equal(expectedSecret.StringData["GF_SECURITY_ADMIN_USER"]))
+			})
+			It("should create JMX secret and set owner", func() {
+				expectJMXSecret(client, *controller, false)
+			})
+			It("should create Grafana service and set owner", func() {
+				service := &corev1.Service{}
+				err := client.Get(context.Background(), types.NamespacedName{Name: "containerjfr-grafana", Namespace: "default"}, service)
+				Expect(err).To(HaveOccurred())
+
+				reconcileFully(client, *controller, false)
+
+				err = client.Get(context.Background(), types.NamespacedName{Name: "containerjfr-grafana", Namespace: "default"}, service)
+				Expect(err).ToNot(HaveOccurred())
+
+				expectedService := resource_definitions.NewGrafanaService(test.NewContainerJFR())
+				checkMetadata(&service.ObjectMeta, &expectedService.ObjectMeta)
```

Comment:
```suggestion
				checkMetadata(service, expectedService)
```

Reply:
Just passing the objects works for me. Could you do this with the other calls to checkMetadata as well?

Reply:
Done!